drivers/i2c/i2c_shell.c

```diff
@@ -289,6 +291,12 @@ static int cmd_i2c_speed(const struct shell *shell_ctx, size_t argc, char **argv
 	}
 
 	speed = strtol(argv[ARGV_DEV + 1], NULL, 10);
+
+	if (speed > I2C_SPEED_DT) {
```

Comment:
i2c shell command: i2c speed `<device>` `<input>`
will result in setting up the same speed either if we pass input as (1 or 9), (2 or 10), (3 or 11)..
Because in i2c speed shell function, I2C_SPEED_SET(speed) macro masks the speed parameter with 0x7.

Thus, introduced the check for the input parameter as passing the value more than 0x7 is not possible.